console.go

```diff
@@ -132,6 +134,14 @@ func (w ConsoleWriter) Write(p []byte) (n int, err error) {
 	if err != nil {
 		return n, err
 	}
+
+	if w.FormatExtra != nil {
+		err = w.FormatExtra(evt, buf)
+		if err != nil {
+			return n, err
+		}
+	}
```

Comment:
Shouldn't this be called before the \n is added so it can be added to the same line if wanted?

Reply:
Sure. My use case was that I add more lines, so this was simpler to do because additional lines (with stack trace) already had newlines.

Maybe the most general approach is to not print `\n` if `w.FormatExtra` is set and is left to `w.FormatExtra` to finalize the line (or lines)?